lib/astutils.cpp

```diff
@@ -3525,3 +3525,51 @@ bool isSizeOfEtc(const Token *tok)
 {
     return Token::Match(tok, "sizeof|typeof|offsetof|decltype|__typeof__ (");
 }
+
+std::string extractVariableName(const Variable *var)
+{
+    if (!var)
+        return "";
+
+    return formatTypeName(var->typeStartToken(), var->typeEndToken()->next());
+}
+
+std::string formatTypeName(const Token *start, const Token *end)
```

Comment:
I would suggest that a method is added to `Variable` class. I guess I'd like the name `getDataTypeString()` better.